aws/logs_monitoring/parsing.py

```diff
@@ -453,6 +453,26 @@ def awslogs_handler(event, context, metadata):
 
     metadata[DD_SERVICE] = get_service_from_tags(metadata)
 
+    cloudwatch_logs_client = boto3.client("logs")
+    response = None
+    try:
+        response = cloudwatch_logs_client.list_tags_log_group(
+            logGroupName=logs["logGroup"]
+        )
+    except Exception as e:
+        logger.exception(f"Failed to get log group tags due to {e}")
+    if response is not None:
+        formatted_tags = [
+            "{key}:{value}".format(key=k, value=v) if v else k
```

Comment:
I don't really get this ternary operator. Why would we set this to `k` if `not v`?

Reply:
This is to capture how AWS tags work with keys and values. If the value isn't present we don't want to send "<KEY>:", we want to just send "<KEY>"

Reply:
It's confusing indenting, but the list comprehension saying for each item, format if v exists, else just return k

`format(k, v) if v else k for k, v in items()`.

Reply:
I think I understand whats going on but not why

So in this case, is `k` the entire tag?  If its a bad format, I'd think we'd exclude it

Reply:
Yeah, k is the whole tag in this case. It's just like tags in datadog without a ":" - a "just key" tag as opposed to a "key:value" tage

Reply:
Are the tags cached? There could be a lot of calls to retrieve the log group tags - resulting in a lot of errors - any thought to handling HTTP 429 errors and providing a backoff/retry strategy?